install-dev/controllers/http/smarty_compile.php

```diff
@@ -25,21 +25,28 @@
  */
 
 define('_PS_DO_NOT_LOAD_CONFIGURATION_', true);
+$directories = [];
```

Comment:
What is the connection between clear the rights folders in `var/cache` and this update of `smarty` compilation?

Reply:
Because it is done in the same time! And many people can't install PrestaShop because of that:
Uncaught Symfony\Component\Filesystem\Exception\IOException: Failed to remove directory "var/cache/prod/": rmdir(var/cache/prod/): Directory not empty. in vendor/symfony/symfony/src/Symfony/Component/Filesystem/Filesystem.php:180

Reply:
You don't answer my question ... why do we have to update the smarty compilation if the issue is with the clear of the cache?

What is the link between the update of this file and the exception you're talking about?

Reply:
This is undirectly related (this is this controller which create the compiled files into the installer). The person who had add the admin-dev/themes/new-theme/template directory has forgot to pre-compile it in the installer.